docs/guide/dev/integration/react/v2/README.md

```diff
@@ -0,0 +1,526 @@
+
+Example usage:
+
+```js
+import React from 'react';
+import {
+	useCKEditor,
+	CKEditorEventAction,
+	registerEditorEventHandler
+} from 'ckeditor4-react';
+
+function Editor( { someProp } ) {
+	const [ element, setElement ] = React.useState();
+
+	const { editor } = useCKEditor( {
+		element,
+		// `dispatchEvent` is memoized, so initial value of `someProp` will be used
+		dispatchEvent: ( { type } ) => {
+			if ( type === CKEditorEventAction.focus ) {
+				console.log(
+					`Will be called with initial value of ${ someProp }.`
+				);
+			}
+		},
+		subscribeTo: [ 'focus' ]
+	} );
+
+	React.useEffect( () => {
+		if ( editor ) {
+			// Registers new handler with high priority whenver value of `someProp` changes.
+			const cleanup = registerEditorEventHandler( {
+				editor,
+				evtName: 'focus',
+				handler: () => {
+					console.log(
+						`Will be called with current value of ${ someProp } before regular event handlers.`
+					);
+				},
+				priority: 0
+			} );
+			return cleanup;
+		}
+	}, [ editor, someProp ] );
+
+	return <div ref={setElement} />;
+}
+
+export default Editor;
+```
+
+### prefixEventName
+
+Prefixes event name with `__CKE__`. Useful for prefixing custom events in `reducer`.
+
+```js
+import { prefixEventName } from 'ckeditor4-react';
+
+function reducer( state, action ) {
+	switch ( action.type ) {
+		case prefixEventName( 'myCustomEvent' ):
+		// return new state
+	}
+}
+```
+
+### stripPrefix
+
+Removes prefix from event name.
+
+```js
+import { stripPrefix } from 'ckeditor4-react';
+
+const evtName = stripPrefix( CKEditorEventAction.focus );
+
+console.log( evtName === 'focus' ); // true
+```
+
+## TypeScript support
+
+React integration provides TypeScript definitions for all its interfaces. Please note that this library **does not** provide types for `CKEDITOR` namespace. Therefore, `CKEDITOR` object (as well as instances of `editor`, etc.) has `any` type.
+
+Only TypeScript 3.5+ is supported.
+
+Type argument must be provided in order to use custom events.
+
+Usage with `CKEditor` component:
+
+```js
+import * as React from 'react';
+import { CKEditor, CKEditorEventHandler } from 'ckeditor4-react';
+
+function Editor() {
+	return (
+		<CKEditor<{
+			onCustomEvent: CKEditorEventHandler<'customEvent'>;
+		}>
+			onCustomEvent={( { name } ) => {
+				console.log( name ); // 'customEvent'
+			}}
+		/>
+	);
+}
+
+export default Editor;
+```
+
+Usage with `useCKEditor` hook:
+
+```js
+import * as React from 'react';
+import { useCKEditor, prefixEventName } from 'ckeditor4-react';
+
+function Editor() {
+	const [ element, setElement ] = React.useState<HTMLDivElement | null>( null );
+
+	useCKEditor<'customEvent' | 'anotherCustomEvent'>( {
+		element,
+		dispatchEvent: ( { type } ) => {
+			if ( type === prefixEventName( 'customEvent' ) ) {
```

Comment:
I'm thinking about:
- `prefixEventName( 'customEvent' )` should return `__CKE__customEvent`.
- So if `type` is equal to the `prefixEventName( 'customEvent' )`,  `console.log( type );` should return `__CKE__customEvent`, not `customEvent` ?

If I'm not missing something here, please correct strings placed as comments after both `console.log`-s.

Reply:
Indeed, that's my mistake.